lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -284,6 +285,30 @@ impl ShardHolder {
             }
         };
 
+        // Cleanup existing shards if resharding down
+        if is_in_progress && direction == ReshardingDirection::Down {
+            for (&id, shard) in self.shards.iter() {
+                // Skip shards that does not belong to resharding shard key
+                if self.shard_id_to_key_mapping.get(&id) != shard_key.as_ref() {
+                    continue;
+                }
+
+                // Skip target shard
+                if id == shard_id {
+                    continue;
+                }
+
+                // We only cleanup local shards
+                if !shard.is_local().await {
+                    continue;
+                }
+
+                let filter = self.hash_ring_filter(id).expect("hash ring filter");
+                let filter = Filter::new_must(Condition::CustomIdChecker(Arc::new(filter)));
+                shard.cleanup_local_shard(filter).await?;
```

Comment:
Shards may also be left in `Resharding` state, and it is probably best to 'clean that up' here too.

I imagine that if the shard is local and in resharding state, we propose to consensus to activate the shard.

Reply:
Good point. I'll take a look how to add this.

Reply:
What will be annoying is that we don't guarantee the state is actually applied through consensus.

I think we can add a fallback for this in cluster manager itself: if there are resharding shards but there is no resharding operation, clean things up.

Reply:
Done.